server/src/internalClusterTest/java/org/elasticsearch/repositories/RepositoriesDuplicatedCreationIT.java

```diff
@@ -0,0 +1,146 @@
+import org.elasticsearch.plugins.RepositoryPlugin;
+import org.elasticsearch.snapshots.mockstore.MockRepository;
+import org.elasticsearch.test.ESIntegTestCase;
+import org.elasticsearch.xcontent.NamedXContentRegistry;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
+import static org.elasticsearch.test.hamcrest.ThrowableAssertions.assertThatException;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.isA;
+
+@ESIntegTestCase.ClusterScope(supportsDedicatedMasters = false, numDataNodes = 3)
+public class RepositoriesDuplicatedCreationIT extends ESIntegTestCase {
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return Collections.singletonList(UnstableRepository.Plugin.class);
+    }
+
+    public static class UnstableRepository extends MockRepository {
+        public static final String TYPE = "unstable";
+        public static final Setting<List<String>> UNSTABLE_NODES = Setting.stringListSetting(
+            "repository.unstable_nodes",
+            Setting.Property.NodeScope,
+            Setting.Property.Dynamic
+        );
+
+        public UnstableRepository(
+            RepositoryMetadata metadata,
+            Environment environment,
+            NamedXContentRegistry namedXContentRegistry,
+            ClusterService clusterService,
+            BigArrays bigArrays,
+            RecoverySettings recoverySettings
+        ) {
+            super(metadata, environment, namedXContentRegistry, clusterService, bigArrays, recoverySettings);
+            List<String> unstableNodes = UNSTABLE_NODES.get(metadata.settings());
+            if (unstableNodes.contains(clusterService.getNodeName())) {
+                throw new RepositoryException(metadata.name(), "Failed to create repository: current node is not stable");
+            }
+        }
+
+        public static class Plugin extends org.elasticsearch.plugins.Plugin implements RepositoryPlugin {
+            @Override
+            public Map<String, Factory> getRepositories(
+                Environment env,
+                NamedXContentRegistry namedXContentRegistry,
+                ClusterService clusterService,
+                BigArrays bigArrays,
+                RecoverySettings recoverySettings
+            ) {
+                return Collections.singletonMap(
+                    TYPE,
+                    (metadata) -> new UnstableRepository(metadata, env, namedXContentRegistry, clusterService, bigArrays, recoverySettings)
+                );
+            }
+
+            @Override
+            public List<Setting<?>> getSettings() {
+                return List.of(UNSTABLE_NODES);
+            }
+        }
+    }
+
+    public void testDuplicateCreateRepository() throws Exception {
```

Comment:
Why "duplicate"? How about this name?

```suggestion
    public void testCreateInvalidRepository() throws Exception {
```

(similarly in the class name and the repository name)

Reply:
I named it `duplicate` because NPE happend when we put repository for the second time.